diff --git a/src/ipa/nxp/cam_helper/meson.build b/src/ipa/nxp/cam_helper/meson.build
index 9a1bcd34..0c675776 100644
--- a/src/ipa/nxp/cam_helper/meson.build
+++ b/src/ipa/nxp/cam_helper/meson.build
@@ -10,6 +10,7 @@ nxp_ipa_cam_helper_sources = files([
     'camera_helper_os08a20.cpp',
     'camera_helper_ov5640.cpp',
     'camera_helper_ox05b1s.cpp',
+    'camera_helper_imx477.cpp',
     'md_parser_ox.cpp',
 ])
 
diff --git a/src/ipa/nxp/neo/data/meson.build b/src/ipa/nxp/neo/data/meson.build
index eaea4bea..b147505c 100644
--- a/src/ipa/nxp/neo/data/meson.build
+++ b/src/ipa/nxp/neo/data/meson.build
@@ -5,9 +5,10 @@ conf_files = files([
     'os08a20.yaml',
     'ov5640.yaml',
     'ox05b1s.yaml',
+    'imx477.yaml',
     'uncalibrated.yaml',
 ])
 
 install_data(conf_files,
              install_dir : ipa_data_dir / 'nxp' / 'neo',
-             install_tag : 'runtime')
\ No newline at end of file
+             install_tag : 'runtime')
