diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index a11846b86fb2..b5fde7789a51 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -707,6 +707,11 @@ imx95-19x19-evk-adv7535-ap1302-dtbs := imx95-19x19-evk.dtb \
 				       imx95-19x19-evk-ap1302.dtbo
 dtb-$(CONFIG_ARCH_MXC) += imx95-19x19-evk-adv7535-ap1302.dtb
 
+imx95-19x19-evk-isp-imx258-adv7535-dtbs := imx95-19x19-evk-neoisp.dtb \
+					   imx95-19x19-evk-imx258.dtbo \
+					   imx95-19x19-evk-adv7535.dtbo
+dtb-$(CONFIG_ARCH_MXC) += imx95-19x19-evk-isp-imx258-adv7535.dtb
+
 imx95-19x19-evk-lt9611uxc-dtbs := imx95-19x19-evk.dtb imx95-19x19-evk-lt9611uxc.dtbo
 dtb-$(CONFIG_ARCH_MXC) += imx95-19x19-evk-lt9611uxc.dtb
 
diff --git a/arch/arm64/configs/imx_v8_defconfig b/arch/arm64/configs/imx_v8_defconfig
index 31eabe2708ea..6e327093e55b 100644
--- a/arch/arm64/configs/imx_v8_defconfig
+++ b/arch/arm64/configs/imx_v8_defconfig
@@ -619,6 +619,7 @@ CONFIG_VIDEO_MHDP_HDMIRX=y
 CONFIG_VIDEO_MHDP_HDMIRX_CEC=y
 CONFIG_VIDEO_HANTRO=m
 CONFIG_VIDEO_IMX219=m
+CONFIG_VIDEO_IMX258=m
 CONFIG_VIDEO_OV5640=y
 CONFIG_VIDEO_OV5645=m
 CONFIG_VIDEO_AP1302=m
