diff --git a/cam_helper/meson.build b/cam_helper/meson.build
index 2d8c0d9..29188c8 100644
--- a/cam_helper/meson.build
+++ b/cam_helper/meson.build
@@ -4,6 +4,7 @@ nxp_ipa_cam_helper_sources = files([
     'camera_helper.cpp',
     'camera_helper_mx95mbcam.cpp',
     'camera_helper_os08a20.cpp',
+    'camera_helper_imx219.cpp',
     'md_parser_ox.cpp',
 ])
 
diff --git a/data/config_ipa_uguzzi.yaml b/data/config_ipa_uguzzi.yaml
index 44ac1bc..3b9be7f 100644
--- a/data/config_ipa_uguzzi.yaml
+++ b/data/config_ipa_uguzzi.yaml
@@ -66,4 +66,18 @@ sensors:
     socket-port: 50010
   - entity: os08a20 2-0036
     socket-port: 50011
+  - model: imx219
+    profiles:
+     - resolution: [3280, 2464]
+       bit-depth: 10
+       dtp-file: database_imx219_3280_2464.bin
+       tuning-id: 2010
+       tuning-mode: 0
+     - resolution: [1920, 1080]
+       bit-depth: 10
+       dtp-file: database_imx219_1920_1080.bin
+       tuning-id: 2010
+       tuning-mode: 0
+  - entity: imx219 2-0010
+    socket-port: 50020
 ...
diff --git a/data/meson.build b/data/meson.build
index ef8128b..d672adf 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -3,6 +3,8 @@
 conf_files = files([
     'database_mx95mbcam.bin',
     'database_os08a20.bin',
+    'database_imx219_3280_2464.bin',
+    'database_imx219_1920_1080.bin',
     'config_ipa_uguzzi.yaml',
 ])
 
